evalml/tests/component_tests/test_drop_nan_rows_transformer.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import pandas as pd
+from pandas.testing import assert_frame_equal, assert_series_equal
+from woodwork import init_series
+
+from evalml.pipelines.components.transformers.preprocessing import (
+    DropNaNRowsTransformer,
+)
+from evalml.utils.woodwork_utils import _schema_is_equal
+
+
+def test_drop_rows_transformer():
+    # Expecting float because of np.NaN values
+    X = pd.DataFrame({"a column": [np.NaN, 2, 3], "another col": [4, np.NaN, 6]})
+    X_expected = pd.DataFrame(
+        {"a column": [3], "another col": [6]}, index=[2], dtype=np.float64
+    )
+    drop_rows_transformer = DropNaNRowsTransformer()
+    drop_rows_transformer.fit(X)
+    transformed_X, _ = drop_rows_transformer.transform(X)
+    assert_frame_equal(transformed_X, X_expected)
+
+    drop_rows_transformer = DropNaNRowsTransformer()
+    fit_transformed_X, _ = drop_rows_transformer.fit_transform(X)
+    assert_frame_equal(fit_transformed_X, X_expected)
+
+
+def test_drop_rows_transformer_retain_ww_schema():
+    # Expecting float because of np.NaN values
+    X = pd.DataFrame(
+        {"a column": [np.NaN, 2, 3, 4], "another col": ["a", np.NaN, "c", "d"]}
+    )
+    X.ww.init()
+    X.ww.set_types(
+        logical_types={"another col": "PersonFullName"},
+        semantic_tags={"a column": "custom_tag"},
+    )
+
+    X_expected = pd.DataFrame({"a column": [3], "another col": ["c"]}, index=[2])
+    X_expected = X_expected.astype({"a column": "float", "another col": "string"})
+    X_expected_schema = X.ww.schema
+
+    y = pd.Series([3, 2, 1, np.NaN])
+    y = init_series(y, logical_type="IntegerNullable", semantic_tags="y_custom_tag")
+
+    y_expected = pd.Series([True], index=[2])
```

Comment:
OK, so why does this need to be `True`?  I assume Woodwork is helping us out and casting this to an integer, but can't y_expected be 1?

And a check on my understanding of the drop nan rows transformer is that we want to drop rows where the target is NaN for sure and, I guess, rows where there are NaN values in the columns.  I would expect then that we lose the 1st row due to "a column", the 2nd row due to "another col" and the last row due to the target.  That leaves the 3rd row, index 2.  Did I get that right?

Reply:
I think the conversion to IntegerNullable in `init_series` will convert it to the correct value (1) but I agree it's confusing to have ti say `True` here.

Reply:
Yeah, this was left over from experimenting with different data types, I intended this to be 1.

@chukarsten your explanation is correct!